src/NuGet.Core/NuGet.Common/NuGetEnvironment.cs

```diff
@@ -94,11 +94,17 @@ public static string GetFolderPath(SpecialFolder folder)
                         {
                             return GetEnvironmentVariable("APPDATA");
                         }
+                        else if (RuntimeEnvironmentHelper.IsLinux)
+                        {
+                            return Path.Combine("etc", "opt");
+                        }
+                        else if (RuntimeEnvironmentHelper.IsOSX)
+                        {
+                            return Path.Combine("Library", "Application Support","NuGet");
+                        }
                         else
                         {
-                            return FirstNonEmpty(
-                                () => GetEnvironmentVariable("XDG_CONFIG_HOME"),
-                                () => Path.Combine(GetHome(), ".config"));
+                            return null;
```

Comment:
null here seems unexpected, it will just crash somewhere higher up. 

How about we use the linux path here? Worst case it won't be found. We could also throw an exception for an unknown OS but that would be very blocking for those users.